dropgz/cmd/root.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	zaplogfmt "github.com/jsternberg/zap-logfmt"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	ctx       context.Context
+	z         *zap.Logger
+	levelFlag string
+	leveler   = zap.NewAtomicLevel()
+)
+
+// root represent the base invocation.
+var root = &cobra.Command{
+	Use:          "dropgz",
+	SilenceUsage: true,
+}
+
+func init() {
+	// set up signal handlers
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithCancel(context.Background())
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-sig
+		cancel()
+		fmt.Println("exiting")
+	}()
```

Comment:
Similar to the other question about `ctx`, this seems like it effectively just swallows SIGTERMs.

Reply:
fixed to exit, since the context is not threaded to timeouts anywhere yet